docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: default
+title:  "Kleisli"
+section: "data"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala"
+scaladoc: "#cats.data.Kleisli"
+---
+# Kleisli
+Kleisli is a data type that will come in handy often, especially if you are working with monadic functions.
```

Comment:
I think it would be nice if the introduction had a bit more "meat" in it, before diving into the detail. Eg something like:

>Kleisli is a data type that will come in handy often, especially if you are working with monadic functions.

>Recall that a Monad is a structure that represents computations and defines how they can be chained together and used to control the flow of the computations.

>But rather than monads, what if we wanted to work with functions with different return types and chain them together? The normal scala compose method cannot be used, so the Kleisli just adds the extra plumbing for us so the functions can work together. For example, we may want to read in different environment configurations and use them in a more general context."